src/components/planner/Schedule.tsx

```diff
@@ -60,6 +62,30 @@ const Schedule = () => {
     return Array.from(aux) as string[]
   }, [classes])
 
+  const slotsOrderedByDay = (slots: Array<SlotInfo>): Array<SlotInfo> => {
+    return slots.sort((slot1, slot2) => (
+      slot1.day - slot2.day
+    ));
+  }
+
+  const SlotBoxes = ({ slots }) => {
+    return slots.map((slot) => {
+      const classDescriptor = classes.find((classDescriptor) => (
+        classDescriptor.classInfo.slots.filter((otherSlot) => otherSlot.id === slot.id).length > 0
+      ))
+
+      if (!classDescriptor) return <></>;
+
+      return <SlotBox
+        courseInfo={classDescriptor.courseInfo}
+        classInfo={classDescriptor.classInfo}
+        classes={classes}
+        hiddenLessonsTypes={hiddenLessonsTypes}
+        slot={slot}
+      />
+    })
+  }
+
```

Comment:
To be honest, I don't like this helper component here. I think it will better for readability not creating and put the `.map` directly in the place.

Also, can we put the `hiddenLessonsTypes` filter here? There's no need to passing to SlotBox

Reply:
I tried avoiding deduplicated code by having this helper component. Maybe it would be a better idea to put this component in a separate file?

We can put the `hiddenLessonTypes` here, yes.

Reply:
I didnt see it in two places. In that case is ok here. You prefer creating a new file be free to decide.

Reply:
Decided to put it into a separate file because it is inefficient to be inside another component due to re-rendering reasons. 

I also removed the `hiddenLessonTypes` from the `SlotBox` and is now inside the `SlotBoxes` like you asked.